xbmc/dbwrappers/dataset.h

```diff
@@ -406,6 +402,49 @@ class Dataset  {
   const sql_record* const get_sql_record();
 
  private:
+ 
+  unsigned int indexMapID;
+
+  struct IndexMapEntry    //Struct to store a indexMapped field access entry
+  {
+   IndexMapEntry(const char *name):resultIndex(~0), strName(name){};
+   bool operator < (const IndexMapEntry &other) const {return strName < other.strName;};
+   unsigned int resultIndex;
+   std::string strName;
+  };
+
+  struct IndexMapComparator
+  {
+   IndexMapComparator(const std::vector<IndexMapEntry> &c):c(c){};
+   bool operator()(const unsigned int &v,const IndexMapEntry &o)const
+   {
+     return c[v] < o;
+   };
+   bool operator()(const unsigned int &v1,const unsigned int &v2)const
+   {
+     return c[v1] < c[v2];
+   };
+   bool operator()(const IndexMapEntry &o,const unsigned int &v)const
+   {
+     return o< c[v];
+   };
+  private:
+   const std::vector<IndexMapEntry> &c;
+  }; 
+
+  std::vector<IndexMapEntry> indexMap_Entries;
+  std::vector<unsigned int> indexMap_Sorter;
+  /* We build up a access list which holds a translation of strings to indices.
+  Assumption for the usage is, that rows of a resultset are normally accessed in 
+  always the same field order.
+  We first look into this list and if we don't get a match we use the
+  slower but more flexible field_value method
+  For the case the retrieval is against our assumption, guessed_sorter is
```

Comment:
Could you please adjust the comment `guessed_sorter` and IMO we should move it over the member declaration.  Sorry that I dont found it a bit earlier